### PR TITLE
Exclude spark from core jar

### PR DESCRIPTION
**Pull request by eneskuluk**
Excludes spark files by executing maven-jar plugin once more to create a jar without spark. After maven-shade creates uber jar, replacing the excluded one with regular jar.

### CODE REVIEW

**Comment by eneskuluk** [comment]
Currently it doesn't include the changes in mvnrepository artifact upload, we haven't talked that yet which was next step, that's why it is currently draft.

**Review by nalinigans — COMMENTED** [review]
Looks good so far @eneskuluk.

**Comment by eneskuluk** [comment]
Added check for gatk in release test by checking skip-gatk-it in tag message. If skip-gatk-it present in tag message, It will skip gatk part, if not it will try to run gatk part. I saw that gatk uses java 17 as default in the test, until java 17 merges on our end, we should skip gatk part always by adding skip-gatk-it to the tag message. If not it will fail right away because it is asking for java 17.

File: .github/workflows/release_publish.yml [hunk 1]
```diff
@@ -57,23 +57,28 @@ jobs:
...
+            -DgroupId=org.genomicsdb -DartifactId=genomicsdb-spark -Dversion=${VERSION_NUMBER} \
+            -Dpackaging=jar -DpomFile=genomicsdb-${VERSION_NUMBER}.pom -DrepositoryId=$REPO_ID \
+            -Djavadoc=genomicsdb-${VERSION_NUMBER}-javadoc.jar \
+            -Dsources=genomicsdb-${VERSION_NUMBER}-sources.jar 
           if [[ ${VERSION_NUMBER} != *SNAPSHOT ]]; then
               mvn nexus-staging:rc-list -DnexusUrl=https://oss.sonatype.org/ -DserverId=ossrh -f genomicsdb-${VERSION_NUMBER}.pom
               stagingRepoId=$(mvn nexus-staging:rc-list -DnexusUrl=https://oss.sonatype.org/ -DserverId=ossrh | grep orggenomicsdb|cut -f2 -d' ')
               echo $stagingRepoId
               mvn nexus-staging:rc-close -DserverId=ossrh -DnexusUrl=https://oss.sonatype.org/ -DstagingRepositoryId=$stagingRepoId -f genomicsdb-${VERSION_NUMBER}.pom
+              mvn nexus-staging:rc-close -DserverId=ossrh -DnexusUrl=https://oss.sonatype.org/ -DstagingRepositoryId=$stagingRepoId -f genomicsdb-${VERSION_NUMBER}.pom
```
Comment (mlathara) [inline .github/workflows/release_publish.yml:81]: I think this needs to be `genomicsdb-spark-${VERSION_NUMBER}.pom` based on what I see in the staging repository.

**Review by mlathara — CHANGES_REQUESTED** [review]
I think this mostly looks good -- one small comment/potential change

File: .github/workflows/release_publish.yml (hunk 1 above)
Comment (eneskuluk) [reply .github/workflows/release_publish.yml:81]: Ah yes, that's correct. Missed that one somehow. Gonna fix it.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -62,10 +70,11 @@ jobs:
       dylib_artifact: libtiledbgenomicsdb.dylib.${{ github.ref_name }}
 
   test:
-    needs: [release-jar]
+    needs: [build-and-push-mac-dylib, release-jar]
```
Comment (nalinigans) [inline .github/workflows/release.yml:73]: I have not understood how this `TAG_MESSAGE` is being used. Shouldn't `release-jar` already be dependent on `build-and-push-mac-dylib`? Do we have to explicitly call out the dependency here?

Comment (eneskuluk) [reply .github/workflows/release.yml:73]: Tag message is used for skipping or not skipping the gatk tests which @mlathara asked me to add. Dependency explicitly added to so we can use output from that step to pass tag message to test step.

**Review by mlathara — APPROVED** [review]
lgtm

**Comment by nalinigans** [comment]
lgtm

**Review by nalinigans — APPROVED** [review]
LGTM

**Pull request merged**